relay-general/src/protocol/span.rs

```diff
@@ -47,6 +49,10 @@ pub struct Span {
     #[metastructure(pii = "true")]
     pub data: Annotated<Object<Value>>,
 
+    /// The origin of the span indicates what created the span (see `OriginType` docs).
+    #[metastructure(max_chars = "enumlike")]
+    pub origin: Annotated<OriginType>,
```

Comment:
Another nit: if we could add this before `data` prop, we would live the "grouping" properties at the end of the structure.